config/initializers/cost_centers.rb

```diff
@@ -13,9 +13,14 @@ def load_from_s3(bucket, key)
   return data
 end
 
+def load_dummy_cost_centres()
+  File.read(File.join(Rails.root, 'config', 'cost_centre_fixture.csv'))
+end
+
 def get_cost_centre_data()
-  if Rails.env.development?
-    return File.read(File.join(Rails.root, 'config', 'cost_centre_fixture.csv'))
+  # dont load the cost centers from s3 if were not running the server or we're not in RAILS_ENV != production
+  if Rails.env.development? or !defined?(Rails::Server)
```

Comment:
seems reasonable .. I would maybe double down on making the logic explicit by extracting `!defined?(Rails::Server)` into a named method eg.

```
def running_inside_a_rake_task?
   @defined(Rails::Server)
end

...
def get_cost_centre_data()
...
 if Rails.env.development? or running_inside_a_rake_task?
   ...
 end
...
```

Reply:
Corin made the change suggested. However, Hauwa and I have removed it because the cost_centres initializer only gets run when the rails app is running and not when the tests are running, so this never does anything. We've changed this logic to look at whether the `COST_CENTRE_S3_BUCKET_NAME` is set and load the dummy CSV file is it isn't.